### PR TITLE
Auto clicking countermeasure (cd + cleanup)

### LINKED ISSUE #344: House Dupe
### Expected behaviour
<What should have happened?>
I dont know maybe sell and now close dialog.

<What actually happened?>

If you get an auto clicker you can buy a house, then spam sell the house with an auto clicker and you will get a lot of money.

### [RPT](https://community.bistudio.com/wiki/Crash_Files#Arma_3) and/or extDB3 logs (if applicable)
<Check your client and server RPT and extDB3 logs for errors and warnings.>
<If there are relevant messages and they are short then post them here with code formatting.>
<If there are relevant messages and they are long then upload the whole file to GitHub, Pastebin, or Hastebin.>

http://oi67.tinypic.com/pnox.jpg

Mission version: 5

### PR DESCRIPTION
**Pull request by SimZor**
<Please review the guidelines for contributing to this repository. The link is above.>

Resolves #344 

#### Changes proposed in this pull request: 
- Cleanup house menu
- Fix buying house multiple times

### CODE REVIEW

File: Altis_Life.Altis/core/housing/fn_houseMenu.sqf [hunk 1]
```diff
@@ -92,14 +92,14 @@ if (!(_curTarget in life_vehicles) || isNil {_curTarget getVariable "house_owner
...
 
-    if (isClass (missionConfigFile >> "Housing" >> worldName >> (typeOf _curTarget))) then {
-        _Btn1 ctrlSetText localize "STR_pInAct_BuyHouse";
-        _Btn1 buttonSetAction "[life_pInact_curTarget] spawn life_fnc_buyHouse;";
-        _isHouse = true;
-    } else {
-        _Btn1 ctrlSetText localize "STR_pInAct_BuyGarage";
-        _Btn1 buttonSetAction "[life_pInact_curTarget] spawn life_fnc_buyHouse;";
-    };
+    private _buildingPurchaseString = [
```
Comment (H0lidays) [inline Altis_Life.Altis/core/housing/fn_houseMenu.sqf]: Interesting way to do it +1

Comment (SimZor) [inline Altis_Life.Altis/core/housing/fn_houseMenu.sqf]: ;)

File: Altis_Life.Altis/core/housing/fn_houseMenu.sqf [hunk 2]
```diff
@@ -92,14 +92,14 @@ if (!(_curTarget in life_vehicles) || isNil {_curTarget getVariable "house_owner
...
+        "STR_pInAct_BuyGarage",
+        "STR_pInAct_BuyHouse"
+    ] select (isClass (missionConfigFile >> "Housing" >> worldName >> typeOf _curTarget));
+    
+    _isHouse = (isClass (missionConfigFile >> "Housing" >> worldName >> typeOf _curTarget));
```
Comment (ColinM9991) [inline Altis_Life.Altis/core/housing/fn_houseMenu.sqf]: I'd recommend moving this before line 95, and you can use `_isHouse` in your string selection

**Comment by setoy** [comment]
lgtm

File: Altis_Life.Altis/core/housing/fn_houseMenu.sqf [hunk 3]
```diff
@@ -89,17 +89,14 @@ if (_curTarget isKindOf "House_F" && playerSide isEqualTo west) exitWith {
...
-        _isHouse = true;
-    } else {
-        _Btn1 ctrlSetText localize "STR_pInAct_BuyGarage";
-        _Btn1 buttonSetAction "[life_pInact_curTarget] spawn life_fnc_buyHouse;";
-    };
+    private _isHouse = (isClass (missionConfigFile >> "Housing" >> worldName >> typeOf _curTarget));
+    private _buildingPurchaseString = [
+        "STR_pInAct_BuyGarage",
+        "STR_pInAct_BuyHouse"
+    ] select _isClass;
```
Comment (setoy) [inline Altis_Life.Altis/core/housing/fn_houseMenu.sqf]: `_isClass` is never defined

Comment (SimZor) [inline Altis_Life.Altis/core/housing/fn_houseMenu.sqf]: Too many times writing isClass in SQF ;)

**Comment by setoy** [comment]
lgtm

**Comment by mur202** [comment]
It still works to dupe

**Comment by mur202** [comment]
It still works to autoclick on sell house, I just did it and I got 52mill

**Comment by mur202** [comment]
You autoclick the sellhouse button. That opens up like 20 tabs, then you can click sellhouse on all of them

**Comment by SimZor** [comment]
@mur202 Wouldn't be possible with my new changes, as it closes the dialog BEFORE it does ANYTHING else. Are you sure you have updated to latest version?

**Comment by SimZor** [comment]
My comment above may have come off as a bit arrogant. After speaking with some other developers, there might be some few things in the engine that can cause issues with this still and still have some space/time to click. In my mind it shouldn't be possible since it should be unscheduled and it should close dialog before it moves on.

**Comment by mur202** [comment]
Make a check for just that dialog ID and if there are more of that kind, run some code.

**Comment by SimZor** [comment]
Wouldnt make any difference

**Pull request merged**